### ISSUE TITLE
Problem about hicPCA

### ISSUE THREAD

**Issue by QianzhaoJ** [deeptools/HiCExplorer#346]
For calling compartment ,I tried different methods including **hicPCA**、[**HiTC**](http://www.bioconductor.org/packages/release/bioc/html/HiTC.html)、[**cooltools**](https://github.com/mirnylab/cooltools).
  But I found the results of calling compartment using these methods are different . The results of **HiTC** and **cooltools** are similar, while the results of **hicPCA** is different in some regions of chromosomes especially chr1 and chr2 .
  I don't know which should to be used for subsequent analysis . Can you give me some advices ?
  **hicPCA** 
 `hicPCA --matrix $h5 --o EIG1.bedGraph EIG2.bedGraph --numberOfEigenvectors 2 --geneTrack $gene`
**HiTC** and **cooltools** both used default parameters.

**Comment by joachimwolff** [comment]
Hi,

can you tell me what 'is different' means in detail? Maybe a plot of the region would be good to understand your issue (sth like [this](https://www.nature.com/articles/s41467-017-01724-9/figures/1)). Are the PCA values just flipped? Does the pattern match the checkerboard pattern of the Pearson correlation matrix (like [this](https://www.researchgate.net/figure/Correlation-analysis-demonstrates-that-the-nucleus-is-segregated-into-two-compartments_fig3_44590762))? How does our PCA and the result of HiTC and cooltools correlate to known protein associated with A or B compartment (E.g. CTCF and Cohesin with A compartment and H3K9me3 with B compartment)?  

In general: It depends on the organism and the way the A / B compartments are computed. We use the method proposed by Lieberman-Aiden in 2009. For organisms like Drosophila we recommend to switch the obs/exp computation to 'norm'. 

Best,

Joachim

**Comment by QianzhaoJ** [comment]
Thanks for your quick reply !
the hicPCA looks like 
![image](https://user-images.githubusercontent.com/42069662/52653767-a2075700-2f2b-11e9-9904-a6a415e5e36c.png)
the HiTC looks like
![image](https://user-images.githubusercontent.com/42069662/52653930-edba0080-2f2b-11e9-8a7f-faf3712a2751.png)
They are both flipped by same bedGraph file describing the genes position .
Best 
Qianzhao

**Comment by joachimwolff** [comment]
Hi,

can you plot these in the same image and make sure they represent the same region? You can use hicPlotTADs or pyGenomeTracks for this. I am not sure if the scaling is equal and I need to have a CTCF track next to it the give useful feedback. 

Best,

Joachim

**Comment by LeilyR** [comment]
I would also recommend you to use `--norm` which takes the affect of proximity ligation assay into account, and compare the output.

**Comment by QianzhaoJ** [comment]
Hi 
Thanks for your advices. I have plotted these in one image and  I tries to add '--norm' ,but the problem remains .
In chr1 or chr2 ,The results of **HiTC** and **cooltools** are similar, while the results of **hicPCA** is different .
![image](https://user-images.githubusercontent.com/42069662/52804532-61404700-30bf-11e9-924d-2eb869f29a78.png)
In some chromosomes (like chr10) , the results of three methods are similar
![image](https://user-images.githubusercontent.com/42069662/52804835-2ab6fc00-30c0-11e9-9572-85d400ffcce1.png)
I will add the track of ATAC  or other known protein associated with A or B compartment tomorrow.
Best 
Qianzhao

**Comment by joachimwolff** [comment]
Sometimes it happens that the first eigenvector shows not A/B compartments but the chromosome arms. Please consider the second eigenvector from the PCA for chr1 and chr2.

**Comment by gtrichard** [comment]
Combining different eigenvectors for different chromosomes might not be the good solution.

The simplest would be to discard chromosome arms regions with `hicAdjustMatrix`. Otherwise you could call compartments on the beginning of chr1 and then on the end of chr1 separately.

But still, I don't get why this issue is `hicPCA` specific.

**Comment by QianzhaoJ** [comment]
Hi :
   The results of  `hicPCA` are becoming similar with  the other two methods after I remove the regions of centromere using `hicAdjustMatrix` .
![image](https://user-images.githubusercontent.com/42069662/53064801-129e0d00-3504-11e9-8534-b84d8a1b6a52.png)
  So,maybe `hicPCA` could improve on that .

**Comment by zyqfrog10** [comment]
Hi @gtrichard @joachimwolff , 

As I observed this,
<img width="961" alt="chr22_AB" src="https://user-images.githubusercontent.com/9779738/78945836-441ec280-7a8f-11ea-9ad1-094b0217d0a2.png">

I start to think it might be related to how hicPCA handle zero count regions on the genome in the matrix. 

In the above exploration, I have already removed the centromere. I had no issue with most of the chromosomes except chr13, chr15, and chr22. These three chromosomes' PC1 all look alike. And they all have short arm with sparse or no genes. PC2 looks workable for these three chromosomes. 

But I am wondering if hicPCA can remove zero count regions internally, would the PC1 improve?
Thanks.

-YZ
